src/toolkits/style_transfer/style_transfer_model_definition.cpp

```diff
@@ -0,0 +1,1008 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#include <toolkits/style_transfer/style_transfer_model_definition.hpp>
+
+#include <ml/neural_net/weight_init.hpp>
+#include <toolkits/coreml_export/mlmodel_include.hpp>
+
+namespace turi {
+namespace style_transfer {
+
+using CoreML::Specification::InnerProductLayerParams;
+using CoreML::Specification::NeuralNetwork;
+using CoreML::Specification::NeuralNetworkLayer;
+
+using turi::neural_net::const_weight_initializer;
+using turi::neural_net::float_array_map;
+using turi::neural_net::model_spec;
+using turi::neural_net::ones_weight_initializer;
+using turi::neural_net::zero_weight_initializer;
+using padding_type = model_spec::padding_type;
+
+namespace {
+
+void define_resnet(std::unique_ptr<model_spec>& nn_spec, size_t num_styles) {
```

Comment:
Could the definition be factored out into component substructures or repeated sub blocks? Maybe worth a TODO, no need to do it now

Reply:
It definitely could, I used a Python script to generate this using CoreMLtools so it isn't the most readable code but I just wanted a network I knew would work. I opened an issue to track it: #2453

Reply:
Thanks, although for this kind of thing I actually prefer a TODO to a GH issue. IME, "cleanup" issues don't get seen; but at least TODO items prompt anybody who sees this code to make it better